OpenSim/Simulation/StatesTrajectory.h

```diff
@@ -439,7 +439,9 @@ class OSIMSIMULATION_API StatesTrajectory {
      *      variables in the Model. If true, such columns of the Storage are
      *      ignored.
      * @param enforceConstraints Modify state variable values to satisfy
-     *      kinematic constraints (by calling Model::assemble()).
+     *      kinematic constraints (by calling Model::assemble()). Be careful
+     *      with this option; enforcing constraints can drastically alter the
+     *      provided states if they do not already obey the constraints.
```

Comment:
The comment is helpful, but perhaps the flag should be `performAssembly` instead. Then the comment would say perform the assembly if the provided states are incomplete, e.g. do not provide values for dependent coordinates of constraints.